server/polar/order/repository.py

```diff
@@ -70,6 +70,32 @@ async def get_due_dunning_orders(self, *, options: Options = ()) -> Sequence[Ord
         )
         return await self.get_all(statement)
 
+    async def acquire_payment_lock(self, order_id: UUID) -> bool:
+        """
+        Attempt to acquire a payment lock for an order atomically.
+
+        Returns:
+            True if lock was acquired, False if already locked
+        """
+        from sqlalchemy import update
+
+        result = await self.session.execute(
+            update(Order)
+            .where(Order.id == order_id, Order.payment_lock_acquired_at.is_(None))
+            .values(payment_lock_acquired_at=utc_now())
+        )
+        return result.rowcount > 0
+
+    async def release_payment_lock(self, order_id: UUID) -> None:
+        """Release a payment lock for an order."""
+        from sqlalchemy import update
+
+        await self.session.execute(
+            update(Order)
+            .where(Order.id == order_id)
+            .values(payment_lock_acquired_at=None)
+        )
```

Comment:
Given that we always call this method with an ORM `Order` at hand, I'm wondering if we should implement it like this:

```py
async def release_payment_lock(self, order: Order, *, flush: bool = False) -> Order:
    return await self.update(order, update_dict={"payment_lock_acquired_at": None}, flush=flush)
```

The nice thing with that:

* + We return an `Order` object with the updated `payment_lock_acquired_at` value
* + We can control the flush